### PR TITLE
Making `utils` more consistent

### PR DESCRIPTION
**Pull request by strickvl**
- I moved the `secrets_manager_utils.py` from `utils` directory to the `secrets_managers` directory
- I moved the `utils.py` file from `io` to the main `utils` directory (since it is used in many places in the codebase.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Review by schustmi — APPROVED** [review]

File: src/zenml/integrations/seldon/steps/seldon_deployer.py [hunk 1]
```diff
@@ -35,6 +34,7 @@
     step,
 )
 from zenml.steps.step_context import StepContext
+from zenml.utils import io_utils as io_utils
```
Comment (bcdurak) [inline src/zenml/integrations/seldon/steps/seldon_deployer.py:37]: Perhaps we can remove the `as io_utils` here

File: src/zenml/integrations/tensorflow/materializers/keras_materializer.py [hunk 2]
```diff
@@ -18,8 +18,8 @@
 
 from zenml.artifacts import ModelArtifact
 from zenml.io import fileio
-from zenml.io import utils as fileio_utils
 from zenml.materializers.base_materializer import BaseMaterializer
+from zenml.utils import io_utils as fileio_utils
```
Comment (bcdurak) [inline src/zenml/integrations/tensorflow/materializers/keras_materializer.py:22]: We can now use `io_utils` here for consistency.

**Review by bcdurak — CHANGES_REQUESTED** [review]
Everything looks good, great changes! I left a few small comments for improvement.

File: src/zenml/integrations/tensorflow/materializers/keras_materializer.py (hunk 2 above)
Comment (strickvl) [reply src/zenml/integrations/tensorflow/materializers/keras_materializer.py]: Done. (And caught this in a bunch of other places as well.)

**Review by bcdurak — APPROVED** [review]

**Pull request merged**